dali/test/python/test_pipeline_debug.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from nvidia.dali.pipeline.experimental import pipeline_def
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+import numpy as np
+import os
+from test_utils import get_dali_extra_path
+
+rn50_pipeline_base_debug_values = {}
+file_root = os.path.join(get_dali_extra_path(), 'db/single/jpeg')
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0)
+def rn50_pipeline_base(debug=False):
+    rng = fn.random.coin_flip(probability=0.5, seed=47)
+    if debug:
+        rn50_pipeline_base_debug_values['rng'] = rng.get()
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
+    if debug:
+        rn50_pipeline_base_debug_values['jpegs'] = jpegs.get()
+        rn50_pipeline_base_debug_values['labels'] = labels.get()
+    images = fn.decoders.image(jpegs, device='mixed', output_type=types.RGB)
+    if debug:
+        rn50_pipeline_base_debug_values['images'] = images.get()
+    resized_images = fn.random_resized_crop(images, device="gpu", size=(224, 224), seed=27)
+    if debug:
+        rn50_pipeline_base_debug_values['resized_images'] = resized_images.get()
+    out_type = types.FLOAT16
+
+    output = fn.crop_mirror_normalize(resized_images.gpu(), mirror=rng, device="gpu", dtype=out_type, crop=(
+        224, 224), mean=[0.485 * 255, 0.456 * 255, 0.406 * 255], std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
+    if debug:
+        rn50_pipeline_base_debug_values['output'] = output.get()
+    return rng, jpegs, labels, images, resized_images, output
+
+
+@pipeline_def(batch_size=8, num_threads=3, device_id=0, debug=True)
+def rn50_pipeline():
+    rng = fn.random.coin_flip(probability=0.5, seed=47)
+    print(f'rng: {rng.get().as_array()}')
+    tmp = rng ^ 1
+    print(f'rng xor: {tmp.get().as_array()}')
+    jpegs, labels = fn.readers.file(
+        file_root=file_root, shard_id=0, num_shards=2)
+    if jpegs.get().is_dense_tensor():
+        print(f'jpegs: {jpegs.get().as_array()}')
+    else:
+        print('jpegs shapes:')
+        for j in jpegs.get():
+            print(j.shape())
+    print(f'labels: {labels.get().as_array()}')
+    images = fn.decoders.image(jpegs, device='mixed', output_type=types.RGB)
+    for i in images.get().as_cpu():
+        print(i)
+    for i in images.get():
+        print(i.shape())
```

Comment:
This suggest to me, that we probably need to introduce the `.shape` for TensorList and probably some nice `__repr__` or `__str__` for the DataNodeDebug.
(I think it should be in some of the next steps, not necessarily this PR).